### PR TITLE
Document the precise behavior of VoxelRayTrace::betweenPoints

### PR DESCRIPTION
**Pull request by SOF3**

### CODE REVIEW

File: src/VoxelRayTrace.php [hunk 1]
```diff
@@ -46,11 +48,21 @@ public static function inDirection(Vector3 $start, Vector3 $directionVector, flo
 	 * Performs a ray trace between the start and end coordinates. This returns a Generator which yields Vector3s
 	 * containing the coordinates of voxels it passes through.
 	 *
+	 * The first Vector3 is `$start->floor()`.
+	 * Every subsequent Vector3 has a taxicab distance of exactly 1 from the previous Vector3;
+	 * if the ray crosses the intersection of multiple axis boundaries directly,
+	 * the algorithm prefers crossing the boundaries in the order `Z -> Y -> X`.
+	 *
+	 * If `$end` is on an axis boundary, the final Vector3 may or may not cross that boundary.
```
Comment (dktapps) [inline src/VoxelRayTrace.php:56]: This is fuzzier than I would like. How did you come up with this?

**Review by dktapps — COMMENTED** [review]
Mostly OK, but I have some questions.

File: src/VoxelRayTrace.php (hunk 1 above)
Comment (SOF3) [reply src/VoxelRayTrace.php:56]: by running the function with many test inputs.

File: src/VoxelRayTrace.php [hunk 2]
```diff
@@ -46,11 +48,21 @@ public static function inDirection(Vector3 $start, Vector3 $directionVector, flo
 	 * Performs a ray trace between the start and end coordinates. This returns a Generator which yields Vector3s
 	 * containing the coordinates of voxels it passes through.
 	 *
+	 * The first Vector3 is `$start->floor()`.
+	 * Every subsequent Vector3 has a taxicab distance of exactly 1 from the previous Vector3;
+	 * if the ray crosses the intersection of multiple axis boundaries directly,
+	 * the algorithm prefers crossing the boundaries in the order `Z -> Y -> X`.
```
Comment (dktapps) [inline src/VoxelRayTrace.php:54]: I'm not sure this is accurate. Judging by the code, it should be X first.

Comment (SOF3) [reply src/VoxelRayTrace.php:54]: If all 3 axes are crossed together, tMaxX = tMaxY = tMaxZ, so the last branch is run.

**Comment by dktapps** [comment]
Merging this as-is, but the end case really ought to be investigated further.

File: src/VoxelRayTrace.php (hunk 1 above)
Comment (dktapps) [reply src/VoxelRayTrace.php:56]: do you have any sample data that show this inconsistency?

**Pull request merged**